docs/zh/latest/plugins/datadog.md

```diff
@@ -0,0 +1,147 @@
+# limitations under the License.
+#
+-->
+
+## 简介
+
+`datadog` 是 Apache APISIX 内置的监控插件，可与 [Datadog](https://www.datadoghq.com/)（云应用最常用的监控和可观测性平台之一）无缝集成。`datadog` 插件支持对每个请求和响应周期进行多种指标参数的获取，这些指标参数基本反映了系统的行为和健康状况。
+
+`datadog` 插件通过 UDP 协议将其自定义指标推送给 DogStatsD 服务器，该服务器通过 UDP 连接与 Datadog Agent 捆绑在一起（关于如何安装 Datadog Agent，请参考[Agent](https://docs.datadoghq.com/agent/) ）。DogStatsD 基本上是 StatsD 协议的实现，它为 Apache APISIX Agent 收集自定义指标，并将其聚合成单个数据点，发送到配置的 Datadog 服务器。更多关于 DogStatsD 的信息，请参考 [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent) 。
+
+`datadog` 插件具有将多个指标参数组成一个批处理统一推送给外部 Datadog Agent 的能力，并且可以重复使用同一个数据包套接字。
+
+此功能可以有效解决日志数据发送不及时的问题。在创建批处理器之后，如果对 `inactive_timeout` 参数进行配置，那么批处理器会在配置好的时间内自动发送日志数据。如果不进行配置，时间默认为 5s。
+
+关于 Apache APISIX 的批处理程序的更多信息，请参考 [Batch-Processor](../batch-processor.md#配置)
+
+## 属性
+
+| 名称             | 类型   | 必选项  | 默认值      | 有效值       | 描述                                                                                |
+| -----------      | ------ | -----------  | -------      | -----       | ------------------------------------------------------------                               |
+| prefer_name      | boolean | optional    | true         | true/false  | 如果设置为 `false`，将使用路由/服务的 id 值作为插件的 `route_name`，而不是带有参数的标签名称。   |
+
+该插件支持使用批处理程序来聚集和处理条目（日志/数据）的批次。这就避免了插件频繁地提交数据，默认情况下，批处理程序每 `5` 秒或当队列中的数据达到 `1000` 时提交数据。有关信息或自定义批处理程序的参数设置，请参阅[批处理程序](../batch-processor.md#configuration) 配置部分。
+
+## 元数据
+
+| 名称        | 类型    | 必选项 |     默认值        | 有效值         | 描述                                                            |
+| ----------- | ------  | ----------- |      -------       | -----         | ---------------------------------------------------------------------- |
+| host        | string  | optional    |  "127.0.0.1"       |               | DogStatsD 服务器的主机地址                                      |
+| port        | integer | optional    |    8125            |               | DogStatsD 服务器的主机端口                                         |
+| namespace   | string  | optional    |    "apisix"        |               | 由 APISIX 代理发送的所有自定义参数的前缀。对寻找指标图的实体很有帮助，例如：(apisix.request.counter)。                                        |
+| constant_tags | array | optional    | [ "source:apisix" ] |              | 静态标签嵌入到生成的指标中。这对某些信号的度量进行分组很有用。 |
+
+要了解更多关于如何有效地编写标签，请访问[这里](https://docs.datadoghq.com/getting_started/tagging/#defining-tags)
+
+## 输出指标
+
+Apache APISIX 代理，对于每个请求响应周期，如果启用了 datadog 插件，就会向 DogStatsD 服务器输出以下指标。
+
+| Metric Name               | StatsD Type   | Description               |
+| -----------               | -----------   | -------                   |
+| Request Counter           | Counter       | 收到的请求数量。   |
+| Request Latency           | Histogram     | 处理该请求所需的时间（以毫秒为单位）。 |
+| Upstream latency          | Histogram     | 代理请求到上游服务器直到收到响应所需的时间（以毫秒为单位）。 |
+| APISIX Latency            | Histogram     | APISIX 代理处理该请求的时间（以毫秒为单位）。|
+| Ingress Size              | Timer         | 以字节为单位的请求体大小。 |
```

Comment:
ditto

Reply:
`TIMER metric type in DogStatsD is an implementation of HISTOGRAM metric type (not to be confused with timers in the standard StatsD). It measures timing data only: for example, the amount of time a section of code takes to execute.
`
why the type is Timer? https://docs.datadoghq.com/metrics/dogstatsd_metrics_submission/#timer

Reply:
Thank sir for the great codereview:-)
I looked at the source code of the datadog plugin for an answer (sorry, I am not the author of this plugin).

I found that it does implement sending the request body size to the dogstatsd server.

I think the author just choose the `TIMER metric type in DogStatsD is an implementation of HISTOGRAM metric type ` and design it.

Reply:
So is this a bug of datadog plugin？

Reply:
I think it can really be considered a bug

The problem comes from the naming convention of the plugin author. He did not follow the current standard Datadog StatsD naming convention.

I don't think I can change the meaning of the original code.

Reply:
I have comment the author's code.
let me wait for his reply.

Reply:
@bisakhmondal 
Would you take a look at this? Thanks!